File: python/cheese.py
from datetime import date
from expirable_product import ExpirableProduct
from shippable import Shippable

class Cheese(ExpirableProduct, Shippable):
    def __init__(self, name, price, quantity, expiry_date, weight):
        super().__init__(name, price, quantity, expiry_date)
        self.weight = weight
    
    def get_weight(self):
        return self.weight
    
    def requires_shipping(self):
        return True

File: javascript/Cheese.js
// javascript/Cheese.js
import { ExpirableProduct } from './ExpirableProduct.js';
import { IShippable } from './Shippable.js';

export class Cheese extends ExpirableProduct {
    constructor(name, price, quantity, expiryDate, weight) {
        super(name, price, quantity, expiryDate);
        this.weight = weight;
    }

    getWeight() {
        return this.weight;
    }

    requiresShipping() {
        return true;
    }
}
